starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -79,10 +80,12 @@ public Optional<Inventory> update(String id, Inventory inventory) {
   /**
    * Delete Inventory By Id.
    * @param id Id of Inventory.
-   * @return Deleted Inventory.
+   * @return deleted Inventory
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    
+    Inventory inventory = mongoTemplate.findAndRemove(new Query().addCriteria(Criteria.where("id").is(id)),
+      Inventory.class);
+    return Optional.of(inventory);
```

Comment:
Double check the API for optional here. There is a slight differences in methods that could cause an issue.